website/docs/advanced-concepts/granular-access-control/how-to-guides/restrict-query-access.mdx

```diff
@@ -0,0 +1,76 @@
+- Enabled the Programmatic Access Control feature and have access to read roles assigned to users. For more information, see [Admin Settings - Programmatic Access Control](/getting-started/setup/instance-configuration).
+- Access to edit role configuration and applications.
+
+## Restrict query access and hide widgets
+
+Follow these steps to update your existing role, revoke query permissions, and hide widgets for a better user experience.
+
+### Change role configuration
+
+To allow only specific users to execute a query, identify the query that processes sensitive data (e.g., *query_process_sensitive_data*) and the roles that need access to execute the query, as well as roles that no longer need this permission. Then update the role configuration as follows:
+
+<br/>
+<div style={{ position: "relative", paddingBottom: "calc(50.520833333333336% + 41px)", height: "0", width: "100%" }}>
+<iframe src="https://demo.arcade.software/B0P2DERbKlllmT0BINBJ?embed" frameborder="0" loading="lazy" webkitallowfullscreen mozallowfullscreen allowfullscreen style={{ position: "absolute", top: "0", left: "0", width: "100%", height: "100%", colorScheme: "light" }} title="Appsmith | Create workflow">
+</iframe>
+</div>
+<br/><br/>
+
+1. Navigate to the _Admin Settings_ by clicking the gear (⚙️) icon in the top right corner of your Appsmith Dashboard.
+2. Under **Access Control** in the sidebar, select **Roles**.
+3. Find the role you no longer want to have permission to perform the operation on sensitive data (Role1) from the list of existing roles.
+4. Click the pencil icon next to the role to edit its permissions.
+5. In the _Application Resources_ tab, revoke the **Execute** permission by unchecking the checkbox under **Execute** permission for the relevant query. For example, *query_process_sensitive_data*. By removing execute permissions from roles that no longer need access, you prevent unauthorized data operations.
+6. Click **Save changes** to apply the updated permissions.
+7. Repeat the steps for all roles where you need to revoke access to execute the query.
+
+After saving the changes, the updated roles will no longer have access to perform operations on sensitive data. The application developers can use the `appsmith.user.roles` object to hide widgets.
+
+### Hide widgets
+
+To ensure data security and a smooth user experience, hide widgets from users without the necessary permissions. For example, if a button triggers a sensitive data query, show the button only to users allowed to run that query:
+
+1. Select the button widget in your application.
+2. Enable the **JS** label for **Visible** property in the property pane.
+3. Use the following JavaScript code to set up conditional rendering based on the role that has permission to execute the query. In the below code, the `roles` object holds all the roles assigned to the user. For more information, see [roles](/reference/appsmith-framework/context-object#roles-object). 
+
+   ```javascript
```

Comment:
We should mention that Role1 should give view access to the page where the widget lies. Only then can they see the page and not see the widget. Also, we should use a better name than Role1.

Reply:
This is restricting access to a query - so the permissions are already there. The page already has permission, we are just revoking the permission of the query. 
I will update Role 1.